zerver/tests/tests.py

```diff
@@ -2330,9 +2330,12 @@ def _test_cases(self, tokens, msg_id, body, send_as_user):
         othello = get_user_profile_by_email('othello@zulip.com')
         hamlet = get_user_profile_by_email('hamlet@zulip.com')
         handle_missedmessage_emails(hamlet.id, [{'message_id': msg_id}])
-
+        reply_to_addresses = "noreply@example.com"
         msg = mail.outbox[0]
-        reply_to_addresses = [settings.EMAIL_GATEWAY_PATTERN % (u'mm' + t) for t in tokens]
```

Comment:
I think this should just have a conditional and check that the reply_to address is correct in both cases.  E.g.

`if settings.EMAIL_GATEWAY_PATTERN != '': ... else: reply_to_address = settings.NOREPLY_EMAIL_ADDRESS`